### PR TITLE
Fix gates with nonzero global shift.

### PR DESCRIPTION
**Pull request by sergeisakov**
This PR fixes gates that have "magic" exponents and nonzero global shifts.

### CODE REVIEW

**Review by balopat — CHANGES_REQUESTED** [review]
Hi Sergei, I just saw this and I figured I jump in - thank you for creating this, this seems like a pretty important fix - can you please add unit tests to cover the failing cases?

File: qsimcirq/qsim_circuit.py [hunk 1]
```diff
@@ -29,58 +29,59 @@ def _cirq_gate_kind(gate):
       + 'only 1- or 2-qubit gates are supported.')
   if isinstance(gate, cirq.ops.XPowGate):
     # cirq.rx also uses this path.
-    if gate.exponent == 1:
+    if gate.exponent == 1 and gate.global_shift == 0:
```
Comment (95-martin-orion) [inline qsimcirq/qsim_circuit.py:32]: To simplify this, we could instead add the following block at the start of this function:
```
if isinstance(gate, cirq.EigenGate) and gate.global_shift != 0:
  return NotImplementedError(f'qsim does not support global shift.
      Received gate with global shift of {gate.global_shift}.')
```
(`global_shift` is a property of the `EigenGate` class)

Comment (95-martin-orion) [reply qsimcirq/qsim_circuit.py:32]: I see - I misunderstood. This should be good to go, then.

**Review by 95-martin-orion — APPROVED** [review]

**Comment by sergeisakov** [comment]
> Hi Sergei, I just saw this and I figured I jump in - thank you for creating this, this seems like a pretty important fix - can you please add unit tests to cover the failing cases?

I added some tests. Please check.

**Review by balopat — APPROVED** [review]

**Pull request merged**